homeassistant/components/tuya/config_flow.py

```diff
@@ -31,13 +32,17 @@
 
 # Project Type
 DATA_SCHEMA_PROJECT_TYPE = vol.Schema(
-    {vol.Required(CONF_PROJECT_TYPE, default=0): vol.In(TUYA_PROJECT_TYPE)}
+    {
+        vol.Required(CONF_PROJECT_TYPE, default=TUYA_PROJECT_TYPE_SMART_HOME): vol.In(
+            TUYA_PROJECT_TYPES.keys()
```

Comment:
If we pass a dictionary to `vol.In` the values of the dictionary will be shown to the user in the form and the selected item key returned in the user input.

Reply:
Nope, the keys will show in the dropdown, but when you select, the item will show.
Making that approach confusing for the user looking at the flow.

Reply:
If that's true there's something wrong with the frontend I think. It didn't use to work like that when passing a dict, as far as I recall.

Reply:
I wouldn’t have used time on this if I wasn’t true

Reply:
Can we improve the frontend?

Reply:
Maybe 🤷‍♀️

Reply:
Is it the same frontend behavior if we remove the default and pass a dict?

Reply:
No difference, that’s why I changed the other dictionaries.
PR is closed, don’t waste mor time on it.

Reply:
Ok, I'm just trying to figure out what's not working since this is a pattern we used to support.